### ISSUE TITLE
Extra metric added to model.metrics in TF 2.2

### ISSUE THREAD

**Issue by ageron** [tensorflow/tensorflow#38150]
**System information** 
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): 
No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 
Ubuntu 18.04 (running on Colab)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: 
N/A
- TensorFlow installed from (source or binary):
Running on Colab
- TensorFlow version (use command below): 
2.2.0-rc2
- Python version:
3.6.9
- Bazel version (if compiling from source):
Running on Colab
- GCC/Compiler version (if compiling from source): 
Running on Colab
- CUDA/cuDNN version:
No GPU
- GPU model and memory:
No GPU

**Describe the current behavior**

When I compile my model with `model.compile(..., metrics=["mae"])` I end up with an extra metric in the first position. This breaks my existing code (for example when accessing `model.metrics[0]`, now I get another metric than the one that was excepted).

**Describe the expected behavior**

When I compile the model with N metrics, I expect `model.metrics` to return a list with N metrics, not 1+N.

**Standalone code to reproduce the issue** 

You can run the following code in [this colab](https://colab.research.google.com/drive/1yKy-TC5PyE6ImQemE89SDSiQkFniXIcb).

```python
import numpy as np
from tensorflow import keras

X_train = np.random.rand(100, 10)
y_train = np.random.rand(100, 1)

model = keras.models.Sequential([
    keras.layers.Dense(2, activation="relu", input_shape=[10]),
    keras.layers.Dense(1)
])
model.compile(loss="mse", optimizer="sgd", metrics=["mae", "mse"])
model.fit(X_train, y_train, epochs=2)

assert len(model.metrics) == 2
```

The assertion fails.

**Other info / logs**

This may be related to issue #37990 , but it feels more severe.

There's another behavior change that really confuses me: the `model.metrics` list is empty until `model.fit()` is called.

In short, `model.metrics` seems really broken and unintuitive now.

**Comment by khimraj** [comment]
@ageron, I think there is always one matrix corresponding to loss and there will extra as many as arguments passed in `metrics`.
You can check their names with `model.metrics_names`.
In your example there will be three metrics first for `loss`, second for `mae` and third for `mse`.
Hope it helps you.

**Comment by amahendrakar** [comment]
Was able to reproduce the issue. Please find the gist [here](https://colab.research.google.com/gist/amahendrakar/d7806dd77e70c4bb2dcaa00914f11012/38150.ipynb). Thanks!

**Comment by briannemsick** [comment]
The refactor breaks strike again...

Good luck.

**Comment by pavithrasv** [comment]
@ageron This was an intended change which was part of a refactor as @briannemsick mentioned. I have updated the docs for metrics in https://github.com/tensorflow/tensorflow/commit/a822fa93da3f8e2484df2d030ddbef24b06cbe07

Model loss and output losses are now tracked as part of metrics. Earlier metrics_names included model loss and output losses but the metrics property did not. Now they are consistent.

Thank you!
